source/common/request_source_plugin_impl.cc

```diff
@@ -0,0 +1,118 @@
+  nighthawk::request_source::DummyPluginRequestSourceConfig config;
+  Envoy::MessageUtil::unpackTo(any, config);
+  return std::make_unique<DummyRequestSourcePlugin>(config);
+}
+
+REGISTER_FACTORY(DummyRequestSourceConfigFactory, RequestSourcePluginConfigFactory);
+
+DummyRequestSourcePlugin::DummyRequestSourcePlugin(
+    const nighthawk::request_source::DummyPluginRequestSourceConfig& config)
+    : dummy_value_{config.has_dummy_value()
+                                                 ? config.dummy_value().value()
+                                                 : std::numeric_limits<double>::infinity()} {}
+RequestGenerator DummyRequestSourcePlugin::get() {
+  RequestGenerator request_generator = []() {
+    Envoy::Http::RequestHeaderMapPtr header = Envoy::Http::RequestHeaderMapImpl::create();
+    auto returned_request_impl = std::make_unique<RequestImpl>(std::move(header));
+    return returned_request_impl;
+  };
+  return request_generator;
+}
+
+void DummyRequestSourcePlugin::initOnThread() {}
+
+std::string FileBasedRequestSourceConfigFactory::name() const {
+  return "nighthawk.file-based-request-source-plugin";
+}
+
+Envoy::ProtobufTypes::MessagePtr FileBasedRequestSourceConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::request_source::FileBasedPluginRequestSourceConfig>();
+}
+
+RequestSourcePtr FileBasedRequestSourceConfigFactory::createRequestSourcePlugin(
+    const Envoy::Protobuf::Message& message, Envoy::Api::Api& api) {
+  const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+  nighthawk::request_source::FileBasedPluginRequestSourceConfig config;
+  Envoy::MessageUtil util;
+  util.unpackTo(any, config);
+  RELEASE_ASSERT(api.fileSystem().fileSize(config.file_path()) < config.max_file_size().value(),
+                 "file size must be less than max_file_size");
+  auto temp_list = std::make_unique<nighthawk::client::RequestOptionsList>();
+  {
+    Envoy::Thread::LockGuard lock_guard(file_lock_);
+    if (options_list_.options_size() == 0) {
+      util.loadFromFile(config.file_path(), options_list_,
+                        Envoy::ProtobufMessage::getStrictValidationVisitor(), api, true);
+    }
+    temp_list->CopyFrom(options_list_);
+  }
+  return std::make_unique<FileBasedRequestSourcePlugin>(config, std::move(temp_list));
+}
+
+REGISTER_FACTORY(FileBasedRequestSourceConfigFactory, RequestSourcePluginConfigFactory);
+
+FileBasedRequestSourcePlugin::FileBasedRequestSourcePlugin(
+    const nighthawk::request_source::FileBasedPluginRequestSourceConfig& config,
+    std::unique_ptr<nighthawk::client::RequestOptionsList> options_list)
+    : uri_(config.uri()), file_path_(config.file_path()),
+      options_list_(std::move(options_list)), request_max_(config.num_requests().value()) {}
+
+RequestGenerator FileBasedRequestSourcePlugin::get() {
+  uint32_t counter = 0;
+  request_count_.push_back(counter);
+  uint32_t* lambda_counter = &request_count_.back();
```

Comment:
instead of a raw pointer, I wonder if we could instead do:
```c++
uint32_t& lambda_counter = request_count_.back();
```

that way, we eliminate the possibility of nulling it, which simplifies things slightly for future readers of this code?